homeassistant/components/vizio/media_player.py

```diff
@@ -83,19 +98,44 @@
             "is valid and available, device type is correct%s",
             fail_auth_msg,
         )
-        return
+        raise PlatformNotReady
+
+    entity = VizioDevice(device, name, volume_step, device_type, unique_id)
+    entry.add_update_listener(async_entry_updated)
+
+    # Add entity to hass.data so that update listener can access it
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][entry.unique_id] = entity
```

Comment:
Don't store the entity.

Reply:
How do I access the `entity` from the `ConfigEntry`?

Reply:
I resolved this by dispatching an event on update and listening for the event in each instance of `VizioDevice`